sys/mips/82371AB.c

```diff
@@ -0,0 +1,141 @@
+/* 82371AB PCI-ISA bridge driver */
+
+#define KL_LOG KL_DEV
+#include <sys/mimiker.h>
+#include <sys/device.h>
+#include <sys/rman.h>
+#include <sys/bus.h>
+#include <mips/malta.h>
+#include <sys/pci.h>
+#include <sys/interrupt.h>
+#include <dev/piixreg.h>
+#include <dev/isareg.h>
+#include <sys/libkern.h>
+#include <sys/devclass.h>
+
+typedef struct intel_isa_state {
+  resource_t *io;
+  rman_t io_rman;
+} intel_isa_state_t;
+
+static resource_t *intel_isa_alloc_resource(device_t *dev, res_type_t type,
+                                            int rid, rman_addr_t start,
+                                            rman_addr_t end, size_t size,
+                                            rman_flags_t flags) {
+  assert(dev->bus == DEV_BUS_ISA);
+
+  if (type != RT_IOPORTS) {
+    device_t *idev = BUS_METHOD_PROVIDER(dev->parent, alloc_resource);
```

Comment:
Why not just `bus_alloc_resource`? You have literally inlined the `bus_alloc_resource` procedure.

Reply:
Given the past issues with `bus_activate_resource` and `bus_deactivate_resource` I consider these wrappers to be untrustworthy when used in dispatch scenarios inside drivers.